### PR TITLE
[consensus] remove generic T

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

We already failed to maintain the abstraction of consensus (it's tightly coupled with other systems), and the T brings very little value today

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: config/src/config/safety_rules_config.rs [hunk 1]
```diff
@@ -51,12 +51,4 @@ pub enum SafetyRulesService {
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct RemoteService {
     pub server_address: SocketAddr,
-    pub consensus_type: ConsensusType,
-}
-
-#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
-pub enum ConsensusType {
```
Comment (davidiw) [inline config/src/config/safety_rules_config.rs:58]: hahahaahahahah, I hate you, this was such a pain in the ass

File: consensus/safety-rules/src/process_client_wrapper.rs [hunk 2]
```diff
@@ -3,52 +3,32 @@
...
+    safety_rules: Box<dyn TSafetyRules>,
 }
 
-impl<T: Payload> ProcessClientWrapper<T> {
+impl ProcessClientWrapper {
     pub fn new(backend: SecureBackend) -> Self {
         let server_port = utils::get_available_port();
         let server_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), server_port);
 
-        let type_id = TypeId::of::<T>();
```
Comment (davidiw) [inline consensus/safety-rules/src/process_client_wrapper.rs:39]: more of my pride....

File: consensus/safety-rules/src/safety_rules.rs [hunk 3]
```diff
@@ -6,22 +6,16 @@ use crate::{
...
+    vote::Vote, vote_data::VoteData, vote_proposal::VoteProposal,
 };
 use libra_crypto::{ed25519::Ed25519Signature, hash::HashValue};
 use libra_logger::debug;
 use libra_types::{
     block_info::BlockInfo, epoch_change::EpochChangeProof, ledger_info::LedgerInfo,
     validator_signer::ValidatorSigner, validator_verifier::ValidatorVerifier, waypoint::Waypoint,
 };
-use std::marker::PhantomData;
+
```
Comment (davidiw) [inline consensus/safety-rules/src/safety_rules.rs:18]: double whitespace?

File: consensus/safety-rules/src/safety_rules_manager.rs [hunk 4]
```diff
@@ -48,19 +48,19 @@ pub fn extract_service_inputs(config: &mut NodeConfig) -> (Author, PersistentSaf
     (author, storage)
 }
 
-enum SafetyRulesWrapper<T> {
-    Local(Arc<RwLock<SafetyRules<T>>>),
-    Process(ProcessService<T>),
-    Serializer(Arc<RwLock<SerializerService<T>>>),
-    SpawnedProcess(SpawnedProcess<T>),
-    Thread(ThreadService<T>),
+enum SafetyRulesWrapper {
```
Comment (davidiw) [inline consensus/safety-rules/src/safety_rules_manager.rs:51]: this <T> made me an expert in serde :/

File: consensus/src/block_storage/block_store_test.rs [hunk 5]
```diff
@@ -430,7 +439,7 @@ fn test_empty_reconfiguration_suffix() {
         inserter.create_qc_for_block(a5.as_ref(), None),
         a5.as_ref().timestamp_usecs(),
         5,
-        vec![42],
+        vec![test_txn],
     );
-    assert!(a6.verify_well_formed().is_err());
+    a6.verify_well_formed().unwrap_err();
```
Comment (davidiw) [inline consensus/src/block_storage/block_store_test.rs:444]: yes!

File: consensus/src/block_storage/block_tree.rs [hunk 6]
```diff
@@ -9,31 +9,30 @@ use consensus_types::{
 use libra_crypto::HashValue;
 use libra_logger::prelude::*;
 use mirai_annotations::{checked_verify_eq, precondition};
-use serde::Serialize;
+
```
Comment (davidiw) [inline consensus/src/block_storage/block_tree.rs:12]: why the whitespace?

File: consensus/src/liveness/multi_proposer_test.rs [hunk 7]
```diff
@@ -49,17 +49,27 @@ fn test_multi_proposer() {
     }
     assert_eq!(pe.is_valid_proposer(proposers[another_idx], round), None);
 
-    let good_proposal =
-        Block::new_proposal(1, 1, 1, certificate_for_genesis(), &signers[primary_idx]);
+    let good_proposal = Block::new_proposal(
```
Comment (davidiw) [inline consensus/src/liveness/multi_proposer_test.rs:52]: should we make an empty_proposal?

**Comment by davidiw** [comment]
Ping me for a follow up once it compiles:D

File: consensus/src/liveness/multi_proposer_test.rs (hunk 7 above)
Comment (zekun000) [reply consensus/src/liveness/multi_proposer_test.rs:52]: it just saves one field, we still need to fill in other fields

**Comment by matbd** [comment]
The changes in `testsuite/generate-format` LGTM. Next, we may want to either
(1) merge `consensus.yaml` into `libra.yaml` (i.e. remove Corpus::Consensus and have everything recorded in Libra), or
(2) substract the map entries of `libra.yaml` from `consensus.yaml` before writing the latter on disk.
What do you think? (I can also look into this next, especially for 2)

**Comment by zekun000** [comment]
@matbd yeah I was about to ask should we merge them since large of them overlap. also wondering does yaml support something like import so that we don't have to duplicate and still keep isolations.

**Comment by matbd** [comment]
> also wondering does yaml support something like import so that we don't have to duplicate and still keep isolations.

Currently, the yaml format is meant to be easy to re-generate (in a sorta canonical way) and is not optimized to be written by hand. We do care about readability, though, hence my comment.

File: consensus/src/liveness/leader_reputation.rs [hunk 8]
```diff
@@ -5,7 +5,7 @@ use crate::liveness::proposer_election::{next, ProposerElection};
 use consensus_types::common::{Author, Round};
 use libra_logger::prelude::*;
 use libra_types::block_metadata::{new_block_event_key, NewBlockEvent};
-use serde::export::PhantomData;
+
```
Comment (davidiw) [inline consensus/src/liveness/leader_reputation.rs:8]: whitespace :O

**Pull request closed without merging**